api/src/routes/auth.js

```diff
@@ -26,7 +28,12 @@ router.post('/login', (req, res) => {
   const { username, password } = req.body;
 
   User.findByUsername(username, (err, user) => {
-    if (err || !user || !bcrypt.compareSync(password, user.password)) {
+    if (err) {
+      console.error("Error in User.findByUsername:", err);
+      return res.status(500).send({ error: 'Internal Server Error' });
```

Comment:
instead of using `status().send()` use `sendStatus()`. This uses the status number and returns the status message.
```suggestion
      return res.sendStatus(500);
```

[Docs](https://expressjs.com/en/5x/api.html#res.sendStatus)

Reply:
Sometimes we want to do this and sometimes we dont. Sometimes we want to send messages on why it failed rather than just send a http code and let the front end interrpret it, 500 can be vauge but in other cases, its fine.  So i will pick and choose in which scenerios to do this in